src/rent-vs-mortgage.md

```diff
@@ -366,15 +366,35 @@ const rangeType = view(Inputs.radio(['DYNAMIC', 'FIXED'], { label: 'Colour range
 The `FIXED` colour range option ensures the same fixed range of colours for consistent display across both units and houses, while the `DYNAMIC` option adjusts the range based on the data to maximise the contrast.
 
 ```js
-const dataArray = data.toArray().filter(row => row.property_type === propertyType)
-const sa2Map = new Map(dataArray.map(row => [row.sa2_code.toString(), row]))
-const sa2Codes = new Set(sa2Map.keys())
+const dataArray = data.toArray();
+const sa2s = await FileAttachment('data/sa2_simplified_neoval.geojson').json()
+
+const joinDataIntoGeojson = (propertyType) => {
+  return {...sa2s, features: sa2s.features.flatMap(feature => {
+    const extraProperties = dataArray.find(row => 
+      // we have to convert number sa2_code to string to match the geojson
+      row.sa2_code.toString() === feature.properties.REGION_CODE
+    && row.property_type === propertyType)
```

Comment:
We will probably get away with it, but the loop within loop (`.find` inside `.flatMap`) will be more expensive with more complete data (e.g. if we render the whole of AU once we dig up the right data), which is why I switched to `Map` and `Set`. You could still `Map` by combining the SA2 code and the property type in the key.

Reply:
You could also just do this once and put the data for both types of property rather than just one at a time, then just reference the right one when rendering instead.

Reply:
The slow version is 131ms for both computations - would you prefer a slower-easier to read version or a faster slightly more complex version?

Reply:
I think both are easy to read but don't mind either way; your updated version is a good compromise.